src/main/java/org/skriptlang/skript/bukkit/damagesource/elements/ExprDamageType.java

```diff
@@ -0,0 +1,88 @@
+package org.skriptlang.skript.bukkit.damagesource.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.damage.DamageSource;
+import org.bukkit.damage.DamageType;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.damagesource.DamageSourceExperiment;
+import org.skriptlang.skript.bukkit.damagesource.elements.ExprSecDamageSource.DamageSourceSectionEvent;
+
+@Name("Damage Source - Damage Type")
+@Description({
+	"The type of damage of a damage source.",
+	"Attributes of a damage source cannot be changed once created, only while within the 'custom damage source' section."
+})
+@Example("""
+	set {_source} to a custom damage source:
+		set the damage type to magic
+		set the causing entity to {_player}
+		set the direct entity to {_arrow}
+		set the damage location to location(0, 0, 10)
+	damage all players by 5 using {_source}
+	""")
+@Example("""
+	on death:
+		set {_type} to the damage type of event-damage source
+	""")
+@Since("INSERT VERSION")
+@RequiredPlugins("Minecraft 1.20.4+")
+public class ExprDamageType extends SimplePropertyExpression<DamageSource, DamageType> implements DamageSourceExperiment {
+
+	static {
+		registerDefault(ExprDamageType.class, DamageType.class, "damage type", "damagesources");
```

Comment:
This  will conflict with ExprDamageCause (which should probably not allow `damage type` and only allow `damage cause`)